test/extensions/formatter/cel/cel_test.cc

```diff
@@ -42,29 +42,142 @@ TEST_F(CELFormatterTest, TestNodeId) {
+              ProtoEq(ValueUtil::stringValue("GET")));
 }
 
-TEST_F(CELFormatterTest, TestParseFail) {
+TEST_F(CELFormatterTest, TestFormatNumberValue) {
   auto cel_parser = std::make_unique<CELFormatterCommandParser>();
   absl::optional<size_t> max_length = absl::nullopt;
-  EXPECT_EQ(nullptr,
-            cel_parser->parse("INVALID_CMD", "requests.headers['missing_headers']", max_length));
+  auto formatter = cel_parser->parse("CEL", "request.headers[':method'].size()", max_length);
+  EXPECT_THAT(formatter->formatValueWithContext(formatter_context_, stream_info_),
+              ProtoEq(ValueUtil::stringValue("3")));
+
+  auto typed_formatter =
+      cel_parser->parse("TYPED_CEL", "request.headers[':method'].size()", max_length);
+  EXPECT_THAT(typed_formatter->formatValueWithContext(formatter_context_, stream_info_),
+              ProtoEq(ValueUtil::numberValue(3)));
 }
 
-TEST_F(CELFormatterTest, TestNullFormatValue) {
+TEST_F(CELFormatterTest, TestFormatNullValue) {
   auto cel_parser = std::make_unique<CELFormatterCommandParser>();
   absl::optional<size_t> max_length = absl::nullopt;
-  auto formatter = cel_parser->parse("CEL", "requests.headers['missing_headers']", max_length);
+  auto formatter = cel_parser->parse("CEL", "request.headers.nope", max_length);
   EXPECT_THAT(formatter->formatValueWithContext(formatter_context_, stream_info_),
               ProtoEq(ValueUtil::nullValue()));
+
+  auto typed_formatter = cel_parser->parse("TYPED_CEL", "request.headers.nope", max_length);
+  EXPECT_THAT(typed_formatter->formatValueWithContext(formatter_context_, stream_info_),
+              ProtoEq(ValueUtil::nullValue()));
+}
+
+TEST_F(CELFormatterTest, TestFormatBoolValue) {
+  auto cel_parser = std::make_unique<CELFormatterCommandParser>();
+  absl::optional<size_t> max_length = absl::nullopt;
+  auto formatter = cel_parser->parse("CEL", "request.headers[':method'] == 'GET'", max_length);
+  EXPECT_THAT(formatter->formatValueWithContext(formatter_context_, stream_info_),
+              ProtoEq(ValueUtil::stringValue("true")));
+
+  auto typed_formatter =
+      cel_parser->parse("TYPED_CEL", "request.headers[':method'] == 'GET'", max_length);
+  EXPECT_THAT(typed_formatter->formatValueWithContext(formatter_context_, stream_info_),
+              ProtoEq(ValueUtil::boolValue(true)));
+}
+
+TEST_F(CELFormatterTest, TestFormatDurationValue) {
+  auto cel_parser = std::make_unique<CELFormatterCommandParser>();
+  absl::optional<size_t> max_length = absl::nullopt;
+  auto formatter = cel_parser->parse("CEL", "duration(\"1h30m\")", max_length);
+  EXPECT_THAT(formatter->formatValueWithContext(formatter_context_, stream_info_),
+              ProtoEq(ValueUtil::stringValue("1h30m")));
+
+  auto typed_formatter = cel_parser->parse("TYPED_CEL", "duration(\"1h30m\")", max_length);
+  EXPECT_THAT(typed_formatter->formatValueWithContext(formatter_context_, stream_info_),
+              ProtoEq(ValueUtil::stringValue("5400s")));
```

Comment:
I would describe this output as not ideal, but I do not know that having a high quality CEL duration formatter is really that important.